### PR TITLE
Add new evergreen mobile product pages [fix #10182]

### LINKED ISSUE #10182: Create 5 Firefox Mobile Evergreen Pages
There are 5 evergreen pages for Firefox Mobile that are ready to develop. Each page is based on the layout of an existing page.

Overall brief: https://docs.google.com/document/d/102A3ZzjDf6v_ZWr-bQODBDmEuNxMaBxin1yQb_KV1S4/edit

***

**Page 1:** Mobile Overview Page
**URL:** /firefox/browsers/mobile/ (redirect from /firefox/mobile/)
**Title:** Firefox Mobile Browsers put your privacy first
**Keywords:** Firefox mobile, mobile browser

[Template]( https://www.mozilla.org/en-US/firefox/browsers/)
[Copy](https://docs.google.com/document/d/1zt2Dw70gZ32lFSK4kuFXu8EHevVy1J3PIXROWnoBNm8/edit?usp=sharing)

***

**Page 2:** Firefox iOS
**URL:** /firefox/browsers/mobile/ios/ 
**Title:** Firefox Browsers iOS (iPhone & iPad)
**Keywords:** Firefox ios, firefox iphone, firefox ipad, iphone browser, default browser

[Template](https://www.mozilla.org/en-US/firefox/mobile/)
[Copy](https://docs.google.com/document/d/1YdSDt4VlvRhHmNJNMJl2Rd1tZ2Haj5nvXBXSvsvxJbA/edit?usp=sharing)

***

**Page 3:** Firefox Android
**URL:** /firefox/browsers/mobile/android/ 
**Title:** Firefox Browser Android
**Keywords:** Firefox android, best android browser

[Template](https://www.mozilla.org/en-US/firefox/mobile/)
[Copy](https://docs.google.com/document/d/1qpTYZrumyyP1AfEngXCZoVURL6TwFrNVCI3Ya0g9p3w/edit?usp=sharing)

***

**Page 4:** Firefox Focus
**URL:** /firefox/browsers/mobile/focus/ 
**Title:** Firefox Focus: The privacy browser
**Keywords:** Firefox Focus, best browser for privacy
**NOTE**: This should have a look/feel of focus and will need designer resources to provide direction

[Template](https://www.mozilla.org/en-US/firefox/mobile/)
[Copy](https://docs.google.com/document/d/18TQqIMtXfEY1mg0ampNVRsiTfUas9q9mwOV1XigXKCA/edit?usp=sharing)

***

**Page 5:** Mobile Comparison Page
**URL:** /firefox/browsers/browsers/mobile/compare/ 
**Title:** 5 of the best mobile Browsers in direct comparison
**Keywords:** 

[Template](https://www.mozilla.org/en-US/firefox/browsers/compare/)
[Copy](https://docs.google.com/document/d/1YchZPfFm2uNywfiO6dHZ0o58n0-DOS4vHmIl-deIrp0/edit?usp=sharing)

### PR DESCRIPTION
**Pull request by craigcook**
## Description
Adds new mobile product pages, replacing the old unified mobile page. We now have separate pages for Android and iOS, as well as a new page for Focus, and a new mobile browser comparison page.

The Focus page has a new native-style mobile banner for Focus, and you'll need to use an iOS or Android device to see it.

The old unified page is being kept temporarily as a fallback for the main mobile index page until sufficient locales can update.

## Issue / Bugzilla link
#10182 

## Testing
http://localhost:8000/firefox/browsers/mobile/
http://localhost:8000/firefox/browsers/mobile/android/
http://localhost:8000/firefox/browsers/mobile/ios/
http://localhost:8000/firefox/browsers/mobile/focus/
http://localhost:8000/firefox/browsers/mobile/compare/

Successful test run: https://gitlab.com/mozmeao/www-config/-/pipelines/366822640

### CODE REVIEW

File: bedrock/exp/urls.py [hunk 1]
```diff
@@ -16,7 +16,6 @@
     page('opt-out', 'exp/opt-out.html'),
     page('firefox', 'exp/firefox/index.html', ftl_files=['firefox/home']),
     url(r'^firefox/new/$', views.new, name='exp.firefox.new'),
-    page('firefox/mobile', 'exp/firefox/mobile.html', ftl_files=['firefox/mobile']),
```
Comment (alexgibson) [inline bedrock/exp/urls.py:19]: Is it worth adding a redirect for this URL, incase there are external links to it?

File: bedrock/firefox/templates/firefox/browsers/mobile/compare.html [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+{% endblock %}
+
+{% block sticky_promo %}
+  {% with non_fx_only=True %}
+    {% include '/includes/sticky-promo.html' %}
+  {% endwith %}
+{% endblock %}
+
+{% block js %}
+  {{ js_bundle('firefox_compare_mobile') }}
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/browsers/mobile/compare.html:423]: The JS bundle here is loading the send-to-device form but not initialising it, so the form posts to a blank page.

File: bedrock/firefox/templates/firefox/browsers/mobile/includes/s2d-focus.html [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+{% if LANG in settings.SEND_TO_DEVICE_LOCALES %}
+  <p class="c-inline-form-body">{{ ftl('mobile-focus-send-a-download-link-to-your') }}</p>
+  {{ send_to_device(include_title=False, message_set='fx-focus', class='vertical', id='s2d-primary') }}
+{% else %}
+  <p class="c-inline-form-body">{{ ftl('mobile-focus-scan-the-qr-code-to-get-started') }}</p>
+  {% if LANG == 'de' %}
+  <div class="qr-code-wrapper">
+    <img src="{{ static('img/firefox/browsers/mobile/qr-klar.png') }}" id="focus-qr" width="400" height="400" alt="{{ ftl('mobile-focus-scan-alt-text') }}">
+  </div>
+  {% else %}
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/browsers/mobile/includes/s2d-focus.html:18]: Would this conditional ever be seen? Seems like s2d is available for DE?

File: bedrock/firefox/templates/firefox/faq.html [hunk 4]
```diff
@@ -62,10 +62,10 @@ <h2>{{ ftl('what-is-firefox') }}</h2>
         <h2>{{ ftl('how-do-i') }}</h2>
         <p>{{ ftl('you-can-easily',
               url=url('firefox.new'),
-              url2=url('firefox.windows'),
-              url3=url('firefox.mac'),
-              url4=url('firefox.linux'),
-              url5=url('firefox.mobile.index')
+              url2=url('firefox.new.protocol.download_windows'),
```
Comment (alexgibson) [inline bedrock/firefox/templates/firefox/faq.html:65]: This page is throwing a 500 error for me with `Reverse for 'firefox.new.protocol.download_windows' not found.`

File: l10n/en/firefox/browsers/mobile/compare.ftl [hunk 5]
```diff
@@ -0,0 +1,67 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+### URL: https://www-dev.allizom.org/firefox/browsers/mobile/compare/
+
+# HTML page title
+mobile-compare-5-of-the-best = 5 of the best mobile Browsers in direct comparison
```
Comment (alexgibson) [inline l10n/en/firefox/browsers/mobile/compare.ftl:8]: Should `Browsers` here be lower case?
```suggestion
mobile-compare-5-of-the-best = 5 of the best mobile browsers in direct comparison
```

File: l10n/en/firefox/browsers/mobile/compare.ftl [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+mobile-compare-almost-all-of = Almost all of the mobile browsers compared here allow complete synchronization between mobile and desktop devices.
+mobile-compare-the-good-news = The good news for { -brand-name-iphone } users is that { -brand-name-apple } has recently allowed { -brand-name-iphone } users to choose their own default browser, so now users can sync their devices across platforms if, for example, they install { -brand-name-firefox } on their { -brand-name-iphone } and { -brand-name-windows } PC.
+
+mobile-compare-conclusion = Conclusion
+mobile-compare-based-on-the = Based on the criteria we outlined — privacy, features, and syncing — there’s really only one mobile browser that satisfies all three, and that’s { -brand-name-firefox }. While similar in the features and syncing categories, { -brand-name-firefox } stands apart in privacy, because we feel it’s the essential component that allows you to just enjoy the internet — exactly the way you like it.
+mobile-compare-our-recommendations-aside = Our recommendations aside, finding the browser that’s right for you will always depend on your individual needs and what you value most as you navigate online.
+
+
+
```
Comment (alexgibson) [inline l10n/en/firefox/browsers/mobile/compare.ftl:40]: Nit: remove blank lines?

File: media/css/firefox/browsers/mobile.scss [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+// not yet recognized as valid by style lint, so we abstract it
+// $mq-reduced-motion: 'screen and (prefers-reduced-motion: reduce)';
+
+// main .mzp-l-content {
+//     max-width: 400px;
+
+//     @media #{$mq-md} {
+//         max-width: $content-max;
+//     }
+// }
```
Comment (alexgibson) [inline media/css/firefox/browsers/mobile.scss:133]: Is this needed still?

File: media/js/firefox/browsers/mobile.js [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    var sendToPrimary = document.getElementById('s2d-primary');
+    // var sendToSecondary = document.getElementById('s2d-secondary');
+
+    // if (sendToPrimary && sendToSecondary) {
+    if (sendToPrimary) {
+        var formPrimary = new Mozilla.SendToDevice('s2d-primary');
+        formPrimary.init();
+
+        // var formSecondary = new Mozilla.SendToDevice('s2d-secondary');
+        // formSecondary.init();
```
Comment (alexgibson) [inline media/js/firefox/browsers/mobile.js:21]: Is this needed?

**Review by alexgibson — CHANGES_REQUESTED** [review]
I'll definitely need a second pass at this one I think, but I took a first run through and left some comments. All the pages look great!

The only common design issue I noticed was the main page headlines bleed over the coloured background gradients at mid-viewport sizes.

Screenshot 2021-09-09 at 16-26-36 Firefox Browser Android

**Comment by craigcook** [comment]
Rebased and updated to address most of the comments.

But it seems the `fx-focus` message set no longer exists in Basket, so marking this as Do Not Merge until we can get that sorted. If we can dig up the old translated emails we can set up a new message set for Focus but if we have to start fresh with a new email we might have to think of something else. Maybe we could launch the Focus page with just the QR code.

**Comment by craigcook** [comment]
The email folks are going to rework the email for Focus and rather than wait any longer we've opted to move ahead with just the QR code on the Focus page, and we can add the widget for Focus when it's ready.

**Review by alexgibson — APPROVED** [review]
r+ ☎️

**Pull request merged**